src/LanguageServer/Protocol/Handler/Completion/CompletionResultFactory.cs

```diff
@@ -86,7 +86,9 @@ internal static class CompletionResultFactory
                 ItemDefaults = new LSP.CompletionListItemDefaults
                 {
                     EditRange = capabilityHelper.SupportDefaultEditRange ? ProtocolConversions.TextSpanToRange(defaultSpan, documentText) : null,
-                    Data = capabilityHelper.SupportCompletionListData ? resolveData : null
+                    Data = capabilityHelper.SupportCompletionListData ? resolveData : null,
+                    // By default, all text edits we create include the appropriate whitespace, so tell the client to leave it as-is.
+                    InsertTextMode = LSP.InsertTextMode.AsIs,
```

Comment:
>InsertTextMode = LSP.InsertTextMode.AsIs, [](http://example.com/codeflow?start=20&length=41)

could consider adding a test validating this is set

Reply:
accidentally force pushed, but fixed